### PR TITLE
feat: thorchain savers withdraw

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR handles THORChain savers withdraws - memoless for the time being.

To be handled as stack on top of this one: 

- The status screen doesn't yet show dust/fees. We will need some additional routing logic for that: given at this point the Tx is already ongoing and the quote is final, we should *not* do the same quote-fetching logic we're doing here

- Confirmed working:
  - [ ] BTC
  - [x] BCH
  - [x] ATOM
  - [ ] DOGE
  - [x] LTC
  - [ ] ETH
  - [x] AVAX

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

None, isolated to savers

## Testing

- Enable the savers flag
- Withdraw a percentage of your position that's not close to the max (can be user-input or a percent option)
- Wait for the Tx to be confirmed on-chain
- Reload the app, wait for the position to refresh, and ensure funds are deducted from your DeFi position and back in your wallet

### Engineering

☝🏽 

### Operations

☝🏽 

### BCH

image
image
image
image
image

### LTC

image
image
image
image
image

### ATOM

image
image
image
image
image

### AVAX

image
image
image
image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3583** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3583" target="_blank">Graphite</a> 
    * **PR #3587** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3587" target="_blank">Graphite</a> 
      * **PR #3600** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3600" target="_blank">Graphite</a> 
        * **PR #3598** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3598" target="_blank">Graphite</a> 
          * **PR #3612** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3612" target="_blank">Graphite</a> 
            * **PR #3614** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3614" target="_blank">Graphite</a> 
              * **PR #3626** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3626" target="_blank">Graphite</a> 
                * **PR #3613** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3613" target="_blank">Graphite</a> 
                  * **PR #3629** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3629" target="_blank">Graphite</a> 
                    * **PR #3630** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3630" target="_blank">Graphite</a> 
                      * **PR #3640** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3640" target="_blank">Graphite</a> 
                        * **PR #3643** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3643" target="_blank">Graphite</a>  👈
                        * **PR #3635** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3635" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/3643?utm_source=stack-comment).

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/ThorchainSaversWithdraw.tsx [hunk 1]
```diff
@@ -42,30 +42,23 @@ export const ThorchainSaversWithdraw: React.FC<WithdrawProps> = ({ accountId })
...
     assetNamespace,
-    assetReference: contractAddress,
+    assetReference,
   })
   const asset = useAppSelector(state => selectAssetById(state, assetId))
-  const underlyingAsset = useAppSelector(state => selectAssetById(state, underlyingAssetId))
+  const underlyingAsset = useAppSelector(state => selectAssetById(state, assetId))
   if (!asset) throw new Error(`Asset not found for AssetId ${assetId}`)
-  if (!underlyingAsset) throw new Error(`Asset not found for AssetId ${underlyingAssetId}`)
+  if (!underlyingAsset) throw new Error(`Asset not found for AssetId ${assetId}`)
```
Comment (gomesalexandre) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/ThorchainSaversWithdraw.tsx:55]: ser

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/ThorchainSaversWithdraw.tsx [hunk 2]
```diff
@@ -42,30 +42,23 @@ export const ThorchainSaversWithdraw: React.FC<WithdrawProps> = ({ accountId })
...
-  })
   const assetId = toAssetId({
     chainId,
     assetNamespace,
-    assetReference: contractAddress,
+    assetReference,
   })
   const asset = useAppSelector(state => selectAssetById(state, assetId))
-  const underlyingAsset = useAppSelector(state => selectAssetById(state, underlyingAssetId))
+  const underlyingAsset = useAppSelector(state => selectAssetById(state, assetId))
```
Comment (gomesalexandre) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/ThorchainSaversWithdraw.tsx:53]: Remove

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx [hunk 3]
```diff
@@ -77,101 +97,328 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
...
+  const asset = useAppSelector(state => selectAssetById(state, assetId ?? ''))
+  const marketData = useAppSelector(state => selectMarketDataById(state, assetId ?? ''))
 
   const accountFilter = useMemo(() => ({ accountId }), [accountId])
   const bip44Params = useAppSelector(state => selectBIP44ParamsByAccountId(state, accountFilter))
   const userAddress = useMemo(() => accountId && fromAccountId(accountId).account, [accountId])
 
   if (!asset) throw new Error(`Asset not found for AssetId ${opportunityData?.assetId}`)
-  if (!feeAsset) throw new Error(`Fee asset not found for AssetId ${feeAssetId}`)
+  if (!asset) throw new Error(`Fee asset not found for AssetId ${assetId}`)
```
Comment (gomesalexandre) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx:120]: Remove

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx [hunk 4]
```diff
@@ -77,101 +97,328 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
...
   const feeAssetBalance = useAppSelector(s =>
     selectPortfolioCryptoHumanBalanceByFilter(s, feeAssetBalanceFilter),
   )
 
+  const selectedCurrency = useAppSelector(selectSelectedCurrency)
+
+  // notify
+  const toast = useToast()
+
+  const getAccountAddress: () => Promise<string> = useCallback(async () => {
```
Comment (gomesalexandre) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx:138]: Maybe UTXO terminology

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx [hunk 5]
```diff
@@ -77,101 +97,328 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
...
+    const withdrawInput = await getWithdrawInput()
+    if (!withdrawInput) throw new Error('Error building send input')
+
+    let txId: string
+
+    // Try/catching and evaluating to something in the catch isn't a good pattern usually
+    // In our case, handleSend() catching means that after all our previous checks, building a Tx failed at coinselect time
+    // So we actually send reconciliate a reconciliate Tx, retry the original send within the same block
+    // and finally evaluate to either the original Tx or a falsy empty string
+    try {
```
Comment (gomesalexandre) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx:328]: Pipeline

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx [hunk 6]
```diff
@@ -77,101 +97,328 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
...
+      // 1. Try to deposit from the originally deposited from / highest UTXO balance address
+      // If this is enough, no other Tx is needed
+      txId = await handleSend({
+        sendInput: withdrawInput,
+        wallet: walletState.wallet,
+      })
+    } catch (e) {
```
Comment (gomesalexandre) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx:335]: Safety factor:
-`chainId` check
- ensure it's a coinselect error


This should be brought to the deposit multiTxSend as well

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx [hunk 7]
```diff
@@ -77,101 +97,328 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
...
+      if (!maybeTxId) {
+        throw new Error('Error sending THORCHain savers Txs')
+      }
+
+      contextDispatch({ type: ThorchainSaversWithdrawActionType.SET_TXID, payload: maybeTxId })
       onNext(DefiStep.Status)
     } catch (error) {
-      moduleLogger.error(error, { fn: 'handleConfirm' }, 'handleConfirm error')
+      moduleLogger.debug({ fn: 'handleWithdraw' }, 'Error sending THORCHain savers Txs')
+      // TODO(gomes): UTXO reconciliation in a stacked PR
```
Comment (gomesalexandre) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx:395]: Remove

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx (hunk 6 above)
Comment (gomesalexandre) [reply src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx:335]: TBD in a follow-up PR to bring it to both and not clutter this diff

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx (hunk 7 above)
Comment (gomesalexandre) [reply src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx]: TBD in a follow-up PR for deposits not clutter this diff

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Status.tsx [hunk 8]
```diff
@@ -33,33 +33,15 @@ export const Status = () => {
   const translate = useTranslate()
   const { state, dispatch } = useContext(WithdrawContext)
   const { query, history: browserHistory } = useBrowserRouter<DefiQueryParams, DefiParams>()
-  const { chainId, assetReference } = query
+  const { chainId } = query
 
-  const assetNamespace = 'erc20'
+  const assetId = state?.opportunity?.assetId
+  const feeAssetId = assetId
```
Comment (gomesalexandre) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Status.tsx:39]: Remove

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Status.tsx [hunk 9]
```diff
@@ -165,7 +147,7 @@ export const Status = () => {
                     ? state.withdraw.estimatedGasCrypto
                     : state.withdraw.usedGasFee,
                 )
-                  .div(`1e+${feeAsset.precision}`)
+                  .div(`1e+${asset.precision}`)
```
Comment (gomesalexandre) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Status.tsx:150]: pow pow

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Status.tsx [hunk 10]
```diff
@@ -176,7 +158,7 @@ export const Status = () => {
...
```
Comment (gomesalexandre) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Status.tsx:161]: Call the powlice we have a missing pow

File: src/state/slices/opportunitiesSlice/resolvers/thorchainsavers/types.ts [hunk 11]
```diff
@@ -40,23 +40,43 @@ export type ThorchainSaverPositionResponse = {
   growth_pct: string
 }
 
-export type ThorchainSaversQuoteResponseSuccess = {
+export type ThorchainSaversCommonQuoteResponseSuccess = {
+  expiry: string // TODO(gomes): guard against expired quote
```
Comment (0xdef1cafe) [inline src/state/slices/opportunitiesSlice/resolvers/thorchainsavers/types.ts:44]: 🧠

File: src/state/slices/opportunitiesSlice/resolvers/thorchainsavers/utils.ts [hunk 12]
```diff
@@ -149,6 +154,43 @@ export const getThorchainSaversQuote = async (
   return quoteData
 }
 
+export const getThorchainSaversWithdrawQuote = async (
+  asset: Asset,
```
Comment (0xdef1cafe) [inline src/state/slices/opportunitiesSlice/resolvers/thorchainsavers/utils.ts]: object over higher arity

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Withdraw.tsx [hunk 13]
```diff
@@ -101,50 +96,86 @@ export const Withdraw: React.FC<WithdrawProps> = ({ accountId, onNext }) => {
...
-    [userAddress, opportunityData, assetReference, yearnOpportunity, asset?.precision],
+    [userAddress, assetReference, accountId, opportunityData, asset, chainId, toast, translate],
   )
 
   const handleContinue = useCallback(
     async (formValues: WithdrawValues) => {
-      if (!(userAddress && dispatch)) return
-      // set withdraw state for future use
+      if (!(userAddress && opportunityData && dispatch)) return
+      // set deposit state for future use
```
Comment (0xdef1cafe) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Withdraw.tsx:155]:
```suggestion
      // set withdraw state for future use
```

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx [hunk 14]
```diff
@@ -77,101 +101,380 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
...
+        stakedAmountCryptoBaseUnit: opportunityData.stakedAmountCryptoBaseUnit,
+        rewardsamountCryptoBaseUnit: opportunityData?.rewardsAmountsCryptoBaseUnit?.[0] ?? '0',
+      })
+
+      const quote = await getThorchainSaversWithdrawQuote({ asset, accountId, bps: withdrawBps })
+
+      const { dust_amount, expected_amount_out } = quote
+      const withdrawFee = amountCryptoThorBaseUnit
+        .minus(expected_amount_out)
+        .div(`1e+${THOR_PRECISION}`)
```
Comment (0xdef1cafe) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx:167]: 🙊

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx [hunk 15]
```diff
@@ -77,101 +101,380 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
...
+    state?.withdraw.cryptoAmount,
+  ])
+
+  const getMaybeUtxoAccountAddress: () => Promise<string> = useCallback(async () => {
+    if (!accountId) throw new Error('accountId is undefined')
+    if (!isUtxoChainId(chainId)) return ''
+
+    try {
+      const position = await getThorchainSaversPosition({ accountId, assetId })
+      debugger
```
Comment (0xdef1cafe) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx:188]: out

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx [hunk 16]
```diff
@@ -77,101 +101,380 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
...
+    )
+
+    const withdrawBps = getWithdrawBps({
+      withdrawAmountCryptoBaseUnit: amountCryptoBaseUnit,
+      stakedAmountCryptoBaseUnit: opportunityData?.stakedAmountCryptoBaseUnit,
+      rewardsamountCryptoBaseUnit: opportunityData?.rewardsAmountsCryptoBaseUnit?.[0] ?? '0',
+    })
+    const quote = await getThorchainSaversWithdrawQuote({ asset, accountId, bps: withdrawBps })
+
+    const maybeUtxoAccountAddress = await getMaybeUtxoAccountAddress()
```
Comment (0xdef1cafe) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx:213]: prob need to guard against this not existing at runtime

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx [hunk 17]
```diff
@@ -77,101 +101,380 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
...
+
+    const { expected_amount_out, dust_amount } = quote
+
+    const amountCryptoThorBaseUnit = toThorBaseUnit({
+      valueCryptoBaseUnit: amountCryptoBaseUnit,
+      asset,
+    })
+    const withdrawFee = amountCryptoThorBaseUnit
+      .minus(expected_amount_out)
+      .div(`1e+${THOR_PRECISION}`)
```
Comment (0xdef1cafe) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx:223]: pow

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx [hunk 18]
```diff
@@ -77,101 +101,380 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
...
+
+    if (!quote) throw new Error('Cannot get THORCHain savers withdraw quote')
+
+    return {
+      from: maybeUtxoAccountAddress,
+      cryptoAmount: fromThorBaseUnit(dust_amount).toFixed(asset.precision),
+      asset,
+      to: quote.inbound_address,
+      sendMax: false,
+      accountId: accountId ?? '',
```
Comment (0xdef1cafe) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx:235]:
```suggestion
      accountId,
```

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx [hunk 19]
```diff
@@ -77,101 +101,380 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
...
+        rewardsamountCryptoBaseUnit: opportunityData?.rewardsAmountsCryptoBaseUnit?.[0] ?? '0',
+      })
+
+      const quote = await getThorchainSaversWithdrawQuote({ asset, accountId, bps: withdrawBps })
+
+      if (!quote) throw new Error('Error getting THORChain savers withdraw quote')
+
+      const { dust_amount } = quote
+
+      const accountAddress = await getMaybeUtxoAccountAddress()
```
Comment (0xdef1cafe) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx:328]: guard against this being empty

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx [hunk 20]
```diff
@@ -77,101 +101,380 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
...
+
+      const sendInput: SendInput = {
+        cryptoAmount: fromThorBaseUnit(dust_amount).toFixed(asset.precision),
+        asset,
+        to: quote.inbound_address,
+        from: accountAddress,
+        sendMax: false,
+        accountId: accountId ?? '',
```
Comment (0xdef1cafe) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx:336]:
```suggestion
        accountId,
```

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx [hunk 21]
```diff
@@ -180,13 +483,13 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
 
   const hasEnoughBalanceForGas = useMemo(
     () =>
-      bnOrZero(feeAssetBalance)
-        .minus(bnOrZero(state?.withdraw.estimatedGasCrypto).div(`1e+${feeAsset.precision}`))
+      bnOrZero(assetBalance)
+        .minus(bnOrZero(state?.withdraw.estimatedGasCrypto).div(`1e+${asset.precision}`))
```
Comment (0xdef1cafe) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx:487]: pow

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx [hunk 22]
```diff
@@ -77,101 +101,380 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
...
+      )
+
+      const bps = getWithdrawBps({
+        withdrawAmountCryptoBaseUnit: amountCryptoBaseUnit,
+        stakedAmountCryptoBaseUnit: opportunityData?.stakedAmountCryptoBaseUnit,
+        rewardsamountCryptoBaseUnit: opportunityData?.rewardsAmountsCryptoBaseUnit?.[0] ?? '0',
+      })
+
+      const quote = await getThorchainSaversWithdrawQuote({ asset, accountId, bps })
+      const accountAddress = await getMaybeUtxoAccountAddress()
```
Comment (0xdef1cafe) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx:271]: guard again

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Status.tsx [hunk 23]
```diff
@@ -158,7 +158,7 @@ export const Status = () => {
                     ? state.withdraw.estimatedGasCrypto
                     : state.withdraw.usedGasFee,
                 )
-                  .div(`1e+${asset.precision}`)
+                  .div(bn(10).pow(asset.precision))
                   .toFixed(5)}
                 symbol='ETH'
```
Comment (0xdef1cafe) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Status.tsx:163]: ?

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx (hunk 19 above)
Comment (gomesalexandre) [reply src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx]: d172d88a6cd573fcd24836e27b2ec2a35501ff19

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx (hunk 16 above)
Comment (gomesalexandre) [reply src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx:212]: d172d88a6cd573fcd24836e27b2ec2a35501ff19

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx (hunk 22 above)
Comment (gomesalexandre) [reply src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx]: b95e138ae81d1a80d6d253a4e2ceefe404e4d258

**Comment by gomesalexandre** [comment]
Tested with some UTXOs / Avalanche as an EVM, other UTXOs and Ethereum is functionally the same - since this will be retested with the full flow from a fresh account, and across more PRs to follow-up, opening this one

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**